File: java/src/main/java/com/fengtian/qianer/atcoder/abc/abc157/ABC157_C.java
package com.fengtian.qianer.atcoder.abc.abc157;

import java.util.Scanner;

public class ABC157_C {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        final int N = sc.nextInt();
        final int M = sc.nextInt();

        final int[][] p = new int[M][2];
        for (int i = 0; i < M; i++) {
            int s = sc.nextInt();
            int c = sc.nextInt();
            p[i][0] = s;
            p[i][1] = c;
        }

        for (int i = 0; i < 1000; i++) {
            String str = String.valueOf(i);
            if (str.length() != N) continue;

            boolean ok = true;
            for (int j = 0; j < M; j++) {
                int k = Character.getNumericValue(str.charAt(p[j][0]-1));
                if (k != p[j][1]) ok = false;
            }

            if (ok) {
                System.out.println(i);
                return;
            }
        }

        System.out.println(-1);
    }
}

File: kotlin/src/abc/abc157/ABC157_C.kt
package abc.abc157

import java.util.*

fun main(args: Array<String>) {
    val sc = Scanner(System.`in`)

    val n = sc.nextInt()
    val m = sc.nextInt()

    val s = mutableMapOf<Int, Int>()
    (0 until m).forEach {
        val a = sc.nextInt()
        val b = sc.nextInt()
        if (s[a] != null && b != s[a]) {
            println(-1)
            return
        }
        s[a] = b
    }

    (0 until 1000).forEach { i ->
        val str = i.toString()
        if (str.length !=n) return@forEach

        var ok = true
        for ((t, u) in s.entries) {
            if (str.length < t) return@forEach
            if (str[t-1] != u.toString()[0]) {
                ok = false
            }
        }

        if (ok) {
            println(i)
            return
        }
    }
    println(-1)
}